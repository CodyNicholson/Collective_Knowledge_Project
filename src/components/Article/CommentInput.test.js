//This file was begun using code from Link.react.test.js, from the Jest Testing React Apps Guide
//https://facebook.github.io/jest/docs/en/tutorial-react.html
import React from 'react';
import { CommentInput } from './CommentInput';
import renderer from 'react-test-renderer';

var user = {image: null, username: "john"}

test('Comment Input component test', () => {
  const component = renderer.create(
	   <CommentInput currentUser={user}>
	   </CommentInput>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  //console.log(JSON.stringify(tree, null, 2));


//create input for textarea
//is setBody() called?

//  let tree = component.toJSON();
 // expect(tree).toMatchSnapshot();

//simulate submit

//create mock create comment


});