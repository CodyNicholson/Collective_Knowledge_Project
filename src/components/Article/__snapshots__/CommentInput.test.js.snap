// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Input component test 1`] = `
<form
  className="card comment-form"
  onSubmit={[Function]}
>
  <div
    className="card-block"
  >
    <textarea
      className="form-control"
      id="commentTA"
      onChange={[Function]}
      placeholder="Write a comment..."
      rows="3"
      value=""
    />
  </div>
  <div
    className="card-footer"
  >
    <img
      alt="john"
      className="comment-author-img"
      src={null}
    />
    <button
      className="btn btn-sm btn-primary"
      type="submit"
    >
      Post Comment
    </button>
  </div>
</form>
`;
